beginner_source/nestedtensor_tutorial.py

```diff
@@ -0,0 +1,449 @@
+"""
+Nested Tensors
+===============================================================
+
+Nested tensor is very similar to regular tensor, except for the shape:
+
+* for a regular tensor, each dimension has a size
+
+* for a nested tensor, not all dimensions have regular sizes: some of them are jagged
+
+Nested tensor arises from many application scenarios:
+
+* in NLP, sentences have variable lengthes, so a batch of sentences forms a nested tensor
+
+* in CV, images have variable shapes, so a batch of images forms a nested tensor
+
+For practical reason, conceptually we implement nested tensor
+as a batch of tensors with different shapes,
+i.e. dimension 0 is assumed to be the batch dimension.
```

Comment:
This block of text seems a bit out-of-place here - maybe it's more useful under Nested Tensor Initialization when showing how to create an NT from a list of tensors?

To transition smoothly from the initial introduction to the meat of the tutorial, maybe a fluff line could be added here instead. Something like:
```
In this tutorial, we will demonstrate basic usage of nested tensors and motivate their usefulness
for operating on sequential data of varying lengths with a real-world example.
```